src/machine.rs

```diff
@@ -1723,3 +1723,77 @@ impl<'tcx> Machine<'tcx> for MiriMachine<'tcx> {
         Cow::Borrowed(ecx.machine.union_data_ranges.entry(ty).or_insert_with(compute_range))
     }
 }
+
+/// Trait for callbacks handling asynchronous machine operations.
+///
+/// Callbacks receive a completion state and can perform follow-up actions while
+/// maintaining interpreter invariants. They are executed with mutable access to
+/// the interpreter context.
+///
+/// # Type Parameters
+/// - `'tcx`: Typing context lifetime for the interpreter.
+/// - `T`: Type of argument passed to the callback on completion.
```

Comment:
```suggestion
/// Trait for callbacks handling asynchronous machine operations.
```
There's a lot of text here but it doesn't say much, so IMO it's not worth it.

Btw, are you using LLMs to generate (parts of) your comments? Knowing where you are using LLMs would help me in the review. :) As you know, LLMs are prone to generate plausible-sounding but subtly wrong text, so one has to review them with a different kind of care than human-written comments.